File: ror/public/javascripts/models/techno.js
// Ember model for techno object
var Techno = DS.Model.extend({
  name: DS.attr('string'),
  puppetclass: DS.attr('string'),
  hiera: DS.attr('string'),
  projects: DS.hasMany('project', {async: true})
});

module.exports = Techno;

File: ror/app/models/techno.rb
# Stores details about one system technology (apache / mongo / mysql / nodejs / ...)
#
# @author Eric Fehr (ricofehr@nextdeploy.io, github: ricofehr)
class Techno < ActiveRecord::Base
  belongs_to :technotype
  # Technos are associated with project object
  has_many :project_technos, dependent: :destroy
  has_many :projects, through: :project_technos
  # Technos are associated with vm object
  has_many :supervises, dependent: :destroy
  has_many :vms, through: :supervises

  # some properties are mandatory and must be well-formed
  validates :name, :puppetclass, presence: true
end
